quantumdb-core/src/main/java/io/quantumdb/core/migration/Migrator.java

```diff
@@ -61,10 +61,9 @@ public Migrator(Backend backend) {
 		this.backend = backend;
 	}
 
-	public void migrate(String sourceVersionId, String targetVersionId) throws MigrationException {
+	public void migrate(State state, String sourceVersionId, String targetVersionId) throws MigrationException {
 		log.info("Forking from version: {} to version: {}", sourceVersionId, targetVersionId);
 
-		State state = loadState();
```

Comment:
Double loading of state

Reply:
This might have some side-effects if `migrate()` is called multiple times (ie. previously reloading state from the database before performing a migration, but now not).

Reply:
Since I have only worked with the terminal commands and not with the java code, I have not thought of this possibility. This might be a problem.

Reply:
I had changed this back, but now I get another problem where you can only change the database if a changeset is first included in the metadata of the quantumdb schema. Calling migrate will not migrate because the changelog will get overwritten.